inter/block.go

```diff
@@ -15,12 +15,31 @@ type Block struct {
 	SkippedTxs  []uint32      // indexes of skipped txs, starting from first tx of first event, ending with last tx of last event
 	GasUsed     uint64
 	Root        hash.Hash
+	prevRandao  common.Hash
 }
 
 func (b *Block) EstimateSize() int {
 	return (len(b.Events)+len(b.InternalTxs)+len(b.Txs)+1+1)*32 + len(b.SkippedTxs)*4 + 8 + 8
 }
 
+// GetPrevRandao returns the PrevRandao for current block.
+func (b *Block) GetPrevRandao() common.Hash {
+	if b.prevRandao == (common.Hash{}) {
+		b.computePrevRandao()
+	}
+	return b.prevRandao
+}
+
+// computePrevRandao computes the PrevRandao from transaction hashes.
+func (b *Block) computePrevRandao() {
+	for _, event := range b.Events {
+		for i := 0; i < 24; i++ {
+			// first 8 bytes should be ignored as they are not pseudo-random.
+			b.prevRandao[i+8] = b.prevRandao[i+8] ^ event[i+8]
+		}
+	}
+}
```

Comment:
In the proposal the xor-value over all event-suffices is hashed to expand the 24-byte value into a 32-byte value.